sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -303,7 +315,34 @@ final class DataFrameWriter private[sql](df: DataFrame) {
       conn.close()
     }
 
-    JdbcUtils.saveTable(df, url, table, props)
+    JdbcUtils.saveTable(df, url, table, props, columnMapping)
+  }
+
+  /**
+   * (java-friendly) version of
+   * [[DataFrameWriter.jdbc(String,String,Properties,scala.collection.Map[String,String]):]]
+   */
+  def jdbc(url: String,
+           table: String,
+           connectionProperties: Properties,
+           columnMapping: java.util.Map[String, String]): Unit = {
+    // Convert java Map into scala Map
+    var sColumnMapping: scala.collection.Map[String, String] = null
+    if (columnMapping!=null) {
+        sColumnMapping = columnMapping.asScala
+    }
+    jdbc(url, table, connectionProperties, sColumnMapping)
+  }
+
+  /**
+   * Three parameter version of
+   * [[DataFrameWriter.jdbc(String,String,Properties,scala.collection.Map[String,String]):]]
+   */
+  def jdbc(url: String,
+           table: String,
+           connectionProperties: Properties): Unit = {
+    val columnMapping: scala.collection.Map[String, String] = null
+    jdbc(url, table, connectionProperties, columnMapping)
```

Comment:
Rather than declare a dummy var, I think you can pass `null.asInstanceOf[Map[String, String]]`? or an empty `Map`? I suppose that depends on what the semantics of an empty map should be.